### PR TITLE
py-scikit-sparse: add 0.4.11, 0.4.12

### PR DESCRIPTION
**Pull request by cgcgcg**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-scikit-sparse/package.py [hunk 1]
```diff
@@ -15,11 +15,16 @@ class PyScikitSparse(PythonPackage):
...
     version("0.4.8", sha256="2a224c60da3ef951975242ea777478583d3265efc72db5cfb7861686521a4009")
 
-    depends_on("python@3.6:3.11", type=("build", "link", "run"))
+    depends_on("python@3.6:3.10", type=("build", "link", "run"), when="@0.4.8")
+    depends_on("python@3.6:3.11", type=("build", "link", "run"), when="@0.4.11")
+    depends_on("python@3.6:3.11", type=("build", "link", "run"), when="@0.4.12")
     depends_on("py-setuptools@40.8:", type="build")
-    depends_on("py-cython@0.22:", type="build")
+    depends_on("py-cython@0.22:0.29", type="build", when="@:0.4.11")
+    depends_on("py-cython", type="build", when="@0.4.12")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-scikit-sparse/package.py]:
```suggestion
    depends_on("py-cython@0.22:", when="@0.4.12:", type="build")
```
see [pyproject.toml#L27](https://github.com/scikit-sparse/scikit-sparse/blob/v0.4.12/pyproject.toml#L27)

File: var/spack/repos/builtin/packages/py-scikit-sparse/package.py [hunk 2]
```diff
@@ -15,11 +15,16 @@ class PyScikitSparse(PythonPackage):
...
     maintainers("cgcgcg")
 
+    version("0.4.12", sha256="e6502fea9ba561cfa5491eb222ed2c81c16263d8182a293950db20509c941166")
+    version("0.4.11", sha256="64c61a8777b7c7ba8e1f2bf76bc767f740e6426f1cce2d90f1324b177618e1ca")
     version("0.4.8", sha256="2a224c60da3ef951975242ea777478583d3265efc72db5cfb7861686521a4009")
 
-    depends_on("python@3.6:3.11", type=("build", "link", "run"))
+    depends_on("python@3.6:3.10", type=("build", "link", "run"), when="@0.4.8")
+    depends_on("python@3.6:3.11", type=("build", "link", "run"), when="@0.4.11")
+    depends_on("python@3.6:3.11", type=("build", "link", "run"), when="@0.4.12")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-scikit-sparse/package.py]:
```suggestion
    depends_on("python@3.6:3.11",  when="@0.4.11:", type=("build", "link", "run"))
```

**Review by manuelakuhn — COMMENTED** [review]
Also optional:
- put dependencies of newer versions before those of older versions
- switch the order of the `when` and `type` argument (as we have for most python packages)

File: var/spack/repos/builtin/packages/py-scikit-sparse/package.py (hunk 1 above)
Comment (cgcgcg) [reply var/spack/repos/builtin/packages/py-scikit-sparse/package.py]: Good catch!

File: var/spack/repos/builtin/packages/py-scikit-sparse/package.py (hunk 2 above)
Comment (cgcgcg) [reply var/spack/repos/builtin/packages/py-scikit-sparse/package.py]: The Python packages normally bumps up their version whenever they support a new Python version. So I think I'll keep the two lines for now.

**Review by tldahlgren — APPROVED** [review]
Confirmed the version sha256.

**Review by adamjstewart — APPROVED** [review]

**Comment by cgcgcg** [comment]
Thanks @manuelakuhn @tldahlgren @adamjstewart

**Pull request merged**